packages/firestore/src/core/sync_engine_impl.ts

```diff
@@ -1543,7 +1598,11 @@ export async function syncEngineApplyActiveTargetsChange(
       /* keepPersistedTargetData */ false
     )
       .then(() => {
-        remoteStoreUnlisten(syncEngineImpl.remoteStore, targetId);
+        // A target could have listened to cache only and no remote store connection has been
+        // established by the primary client.
+        if (syncEngineImpl.sharedClientState.isActiveQueryTarget(targetId)) {
```

Comment:
Will there ever be a case when this evaluates to false?

Reply:
I was expecting if we unlisten to cache listener in primary client, it might go through this function. But seems like it will not. Will remove this check.